src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -780,14 +786,25 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
             break;
         }
 
-        // The system request the min/max window size and position
+        // Fix violations of minimum or maximum size
         case WM_GETMINMAXINFO:
         {
-            // We override the returned information to remove the default limit
-            // (the OS doesn't allow windows bigger than the desktop by default)
-            auto* info             = reinterpret_cast<MINMAXINFO*>(lParam);
-            info->ptMaxTrackSize.x = 50000;
-            info->ptMaxTrackSize.y = 50000;
+            const auto maximumSize = contentSizeToWindowSize(getMaximumSize().value_or(Vector2u(10'000, 10'000)), m_handle);
+
+            MINMAXINFO& minMaxInfo      = *reinterpret_cast<PMINMAXINFO>(lParam);
+            minMaxInfo.ptMaxTrackSize.x = maximumSize.x;
+            minMaxInfo.ptMaxTrackSize.y = maximumSize.y;
+            if (getMaximumSize().has_value())
+            {
+                minMaxInfo.ptMaxSize.x = maximumSize.x;
+                minMaxInfo.ptMaxSize.y = maximumSize.y;
+            }
```

Comment:
This part seems redundant.

Reply:
Nevermind, I thought it was doing the same as the line before but ptMaxTrackSize is not ptMaxSize :facepalm: